src/solvers/refinement/string_refinement.cpp

```diff
@@ -2003,31 +2003,15 @@ static optionalt<exprt> find_counter_example(
 typedef std::map<exprt, std::vector<exprt>> array_index_mapt;
 
 /// \related string_constraintt
-class gather_indices_visitort: public const_expr_visitort
-{
-public:
-  array_index_mapt indices;
-
-  gather_indices_visitort(): indices() {}
-
-  void operator()(const exprt &expr) override
-  {
-    if(expr.id()==ID_index)
-    {
-      const index_exprt &index=to_index_expr(expr);
-      const exprt &s(index.array());
-      const exprt &i(index.index());
-      indices[s].push_back(i);
-    }
-  }
-};
-
-/// \related string_constraintt
 static array_index_mapt gather_indices(const exprt &expr)
 {
-  gather_indices_visitort v;
-  expr.visit(v);
-  return v.indices;
+  array_index_mapt indices;
+  std::for_each(expr.depth_begin(), expr.depth_end(), [&](const exprt &expr) { // NOLINT
```

Comment:
I still find that indentation hard to read. But if I'm the only one in that case then it's fine.

Reply:
@allredj I disabled clang-format on these lines to have brackets on a new line. Can you check it looks ok?